File: js/cm-api-v2/PostType.js
// DocSection: cm_api_v2_post_type
// Tip: Find more about JS/TS SDKs at https://docs.kontent.ai/javascript
// Using ES6 syntax
import { ManagementClient, ElementModels } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_API_KEY>'
});

client.addContentType()
  .withData(builder => {
    return {
      name: "Article",
      codename: "my_article",
      external_id: "article",
      content_groups: [
        {
          name: "Article copy",
          external_id: "article-copy",
        },
        {
          name: "Author",
          codename: "author",
        }
      ],
      elements: [
        builder.textElement({
          name: "Article title",
          codename: "title",
          type: ElementModels.ElementType.text,
          content_group: {
            external_id: "article-copy"
          },
        }),
        builder.richtTextElement({
          name: "Article body",
          codename: "body",
          type: ElementModels.ElementType.richText,
          content_group: {
            external_id: "article-copy"
          },
        }),
        builder.richtTextElement({
          name: "Author bio",
          codename: "bio",
          type: ElementModels.ElementType.richText,
          allowed_blocks: [
            "images",
            "text"
          ],
          content_group: {
            codename: "author"
          }
        })
      ]
    };
  })
  .toObservable()
  .subscribe((response) => {
    console.log(response);
  },
    (error) => {
      console.log(error);
    });
// EndDocSection

File: net/management-api-v2/PostType.cs
// DocSection: cm_api_v2_post_type
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_ENVIRONMENT_ID>"
});

var response = await client.CreateContentTypeAsync(new ContentTypeCreateModel
{
    ExternalId = "article",
    Name = "Article",
    Codename = "my_article",
    ContentGroups = new[]
    {
        new ContentGroupModel
        {
            Name = "Article Copy",
            ExternalId = "article-copy",
        },
        new ContentGroupModel
        {
            Name = "Author",
            CodeName = "author",
        }
    },
    Elements = new ElementMetadataBase[]
    {
        new TextElementMetadataModel
        {
            Name = "Article title",
            Codename = "title",
            ContentGroup = Reference.ByCodename("article-copy"),
            DefaultValue = new TextElementDefaultValueModel
            {
                Global = new TypeValue<string>()
                {
                    Value = "This is the default value of the text element."
                }
            }
        },
        new RichTextElementMetadataModel
        {
            Name = "Article body",
            Codename = "body",
            ContentGroup = Reference.ByCodename("article-copy"),
        },
        new RichTextElementMetadataModel
        {
            Name = "Author bio",
            Codename = "bio",
            AllowedBlocks = new HashSet<RichTextBlockType>()
            {
                RichTextBlockType.Images,
                RichTextBlockType.Text
            },
            ContentGroup = Reference.ByCodename("author"),
        },
    }
});
// EndDocSection
